dlpy/applications/enet.py

```diff
@@ -0,0 +1,271 @@
+    bn2 = BN(act='relu')(tconv1)
+
+    conv3 = Conv2d(out_depth, 1, stride=1, act='identity', include_bias=False)(bn2)
+    bn3 = BN(act='relu')(conv3)
+
+    return bn3
+
+
+def ENet(conn, model_table='ENet', n_classes=2, n_channels=3, width=512, height=512, scale=1.0/255,
+         norm_stds=None, offsets=None, random_mutation=None, init=None, random_flip=None, random_crop=None,
+         output_image_type=None, output_image_prob=False):
+    '''
+    Generates a deep learning model with the E-Net architecture.
+
+    Parameters
+    ----------
+    conn : CAS
+        Specifies the connection of the CAS connection.
+    model_table : string, optional
+        Specifies the name of CAS table to store the model.
+    n_classes : int, optional
+        Specifies the number of classes. If None is assigned, the model will
+        automatically detect the number of classes based on the training set.
+        Default: 2
+    n_channels : int, optional
+        Specifies the number of the channels (i.e., depth) of the input layer.
+        Default: 3
+    width : int, optional
+        Specifies the width of the input layer.
+        Default: 512
+    height : int, optional
+        Specifies the height of the input layer.
+        Default: 512
+    scale : double, optional
+        Specifies a scaling factor to be applied to each pixel intensity values.
+        Default: 1.0/255
+    norm_stds : double or iter-of-doubles, optional
+        Specifies a standard deviation for each channel in the input data.
+        The final input data is normalized with specified means and standard deviations.
+    offsets : double or iter-of-doubles, optional
+        Specifies an offset for each channel in the input data. The final input
+        data is set after applying scaling and subtracting the specified offsets.
+    random_mutation : string, optional
+        Specifies how to apply data augmentations/mutations to the data in the
+        input layer.
+        Valid Values: 'none', 'random'
+    init : str
+        Specifies the initialization scheme for convolution layers.
+        Valid Values: XAVIER, UNIFORM, NORMAL, CAUCHY, XAVIER1, XAVIER2, MSRA, MSRA1, MSRA2
+        Default: None
+    random_flip : string, optional
+        Specifies how to flip the data in the input layer when image data is
+        used. Approximately half of the input data is subject to flipping.
+        Valid Values: 'h', 'hv', 'v', 'none'
+    random_crop : string, optional
+        Specifies how to crop the data in the input layer when image data is
+        used. Images are cropped to the values that are specified in the width
+        and height parameters. Only the images with one or both dimensions
+        that are larger than those sizes are cropped.
+        Valid Values: 'none', 'unique', 'randomresized', 'resizethencrop'
+    output_image_type: string, optional
+        Specifies the output image type of this layer.
+        possible values: [ WIDE, PNG, BASE64 ]
```

Comment:
did we add base64 in 3.5? or is it coming with 4.0.1?

Reply:
This is a new feature of 4.0.1. I added this option as DLPy 1.4.1 is aligned with 4.0.1. Please correct me if not.

Reply:
I believe, I just wanted to make sure I am following what's new for 4.0.1